### PR TITLE
update ci to 1.0, 1.4, nightly

### PR DESCRIPTION
**Pull request by ianshmean**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,23 +3,17 @@ os:
   - linux
   - osx
 julia:
-  - 0.7
   - 1.0
-  - 1.1
-  - 1.2
+  - 1.4
```
Comment (timholy) [inline .travis.yml:7]: This could just be `1` so that the latest stable release gets run.

File: .travis.yml [hunk 2]
```diff
@@ -3,23 +3,17 @@ os:
   - linux
   - osx
 julia:
-  - 0.7
```
Comment (timholy) [inline .travis.yml:6]: Removing testing for 0.7 requires that we remove support for 0.7. If I understand the automatic merge guidelines, that constitutes a breaking release so we'd need to go to 2.0.0. But I'm not certain.

File: .travis.yml (hunk 1 above)
Comment (ianshmean) [reply .travis.yml:7]: Nice! 👍🏻

File: .travis.yml (hunk 2 above)
Comment (ianshmean) [reply .travis.yml:6]: Ok, I didn't mean to be breaking. I thought 0.7 would be ok to remove given it was so temporary. But I like robust logic, so I'll add it back for now

**Pull request merged**